nautobot_ssot_aristacv/diffsync/models/nautobot.py

```diff
@@ -47,38 +49,94 @@ def create(cls, diffsync, ids, attrs):
             device_role=device_role_object,
             site=default_site_object,
             name=ids["name"],
+            serial=attrs["serial"] if attrs.get("serial") else "",
         )
-
-        new_device = nautobot.assign_arista_cf(new_device)
-
-        new_device.validated_save()
-
-        if configs.get("apply_import_tag", APPLY_IMPORT_TAG):
+        if PLUGIN_SETTINGS.get("apply_import_tag", APPLY_IMPORT_TAG):
             import_tag = nautobot.verify_import_tag()
             new_device.tags.add(import_tag)
+        try:
             new_device.validated_save()
-
-        return super().create(ids=ids, diffsync=diffsync, attrs=attrs)
+            return super().create(ids=ids, diffsync=diffsync, attrs=attrs)
+        except ValidationError as err:
+            diffsync.job.log_warning(message=f"Unable to create Device {ids['name']}. {err}")
+            return None
 
     def update(self, attrs):
         """Update device object in Nautobot."""
+        dev = OrmDevice.objects.get(id=self.uuid)
+        if "device_model" in attrs:
+            dev.device_type = nautobot.verify_device_type_object(attrs["device_model"])
+        if "serial" in attrs:
+            dev.serial = attrs["serial"]
+        dev.validated_save()
```

Comment:
Should we throw a validationerror in here to be safe?

Reply:
Sure, I've gone ahead and done so just to be on the safe side. It shouldn't be thrown but ya never know.